exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionerTemplate.java

```diff
@@ -286,7 +286,7 @@ public void flush(boolean schemaChanged) throws IOException {
       //      sender has acknowledged the terminate request. After sending the last batch, all further batches are
       //      dropped.
       //   3. Partitioner thread is interrupted due to cancellation of fragment.
-      final boolean isLastBatch = isLast || terminated || Thread.currentThread().isInterrupted();
+      final boolean isLastBatch = isLast || Thread.currentThread().isInterrupted();
```

Comment:
If the sender is not sending the "last batch" message, could we be sure that no receiver is waiting for it ? Like we discussed it seems the MergingReceiver does not behave the same as UnorderedReceiver in the sense that the MR waits for the last batch.